Show "seccomp" in docker info (#20909).

yongtang opened the pull request:
This pull request added a `SecurityOptions` field in the `GET /info` output to show if there is apparmor, seccomp, or selinux suport.

The API changes are updated in the documentation and the update in `GET /info` is covered by the test case in `TestInfoApi`.

This pull request fixes #20909.

Note: a pull request in https://github.com/docker/engine-api will be opened separately.

Signed-off-by: Yong Tang yong.tang.github@outlook.com

justincormack commented:
Sorry haven't had a chance to review this yet, will do soon.

calavera commented:
Seccomp is a security option like AppArmor and SELinux. Neither of them have a specific output in info, why should we differ with that? I think it would be more useful to have a "SecurityOptions" section that shows whatever the user configured.

calavera commented:
Also, changes in `engine-api` need to go in https://github.com/docker/engine-api.

justincormack commented:
For diagnosing issues it is useful to know what is configured, we did discuss in #20909 how to display it. I think a line `SecurityOptions: seccomp selinux` would probably be the most useful. (Especially as libseccomp-golang is not intending to provide any version information it seems).

yongtang commented:
Thanks @calavera @justincormack for the feedback. Let me update the pull request accordingly.

yongtang committed:
Show "seccomp" in docker info (#20909).

This pull request added a `SecurityOptions` field in the `GET /info`
output to show if there is `apparmor`, `seccomp`, or `selinux` suport.

The API changes are updated in the documentation and the update in
`GET /info` is covered by the test case in `TestInfoApi`.

This pull request fixes #20909.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang commented:
Hi @justincormack @calavera @thaJeztah the Jenkins CI finally passed all tests for this PR (SecurityOptions - seccomp/apparmor/selinux).

Let me know if there are any changes I need to make and any comments would be appreciated.

runcom commented on daemon/info.go:
```diff
@@ -67,6 +67,17 @@ func (daemon *Daemon) SystemInfo() (*types.Info, error) {
+		securityOptions = append(securityOptions, "apparmor")
+	}
+	if sysInfo.Seccomp {
+		securityOptions = append(securityOptions, "seccomp")
```
seccomp and apparmor (?) can have a custom profile loaded. I don't see why this is useful w/o showing what's there. This output is just basically telling me that I have seccomp|apparmor|selinux on my system :|

runcom replied:
i.e. we have a bug report mentioning it's using seccomp but we don't have the profile (if it isn't the default)

justincormack replied:
The main aim is so when triaging issues we know what a user has installed, which is very hard to find now.

For seccomp the profile is per container run, so the daemon cannot print it. For apparmor it could be useful, but we dont want to always print it I dont think, it is large.

thaJeztah commented:
LGTM

vdemeester commented:
LGTM 🐮

calavera commented:
LGTM. Moving to docs review.

thaJeztah commented:
Docs LGTM

vdemeester commented:
Docs LGTM 🐯

thaJeztah commented:
relates to https://github.com/docker/docker/pull/21172
